OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyLogic.cs

```diff
@@ -743,9 +746,56 @@ void UpdatePlayerList()
 			tabCompletion.Names = orderManager.LobbyInfo.Clients.Select(c => c.Name).Distinct().ToList();
 		}
 
+		void UpdateDiscordStatus()
+		{
+			var mapTitle = map.Title;
+			var numberOfPlayers = 0;
+			var slots = 0;
+
+			if (!skirmishMode)
+			{
+				foreach (var kv in orderManager.LobbyInfo.Slots)
+				{
+					if (kv.Value.Closed)
+						continue;
+
+					slots++;
+					var client = orderManager.LobbyInfo.ClientInSlot(kv.Key);
+
+					if (client != null)
+						numberOfPlayers++;
+				}
+			}
+
+			if (updateDiscordStatus)
+			{
+				string secret = null;
+				if (orderManager.LobbyInfo.GlobalSettings.Dedicated)
+				{
+					var endpoint = orderManager.Endpoint.GetConnectEndPoints().First();
+					secret = string.Concat(endpoint.Address, "|", endpoint.Port);
+				}
+
+				var state = skirmishMode ? DiscordState.InSkirmishLobby : DiscordState.InMultiplayerLobby;
+				DiscordService.UpdateStatus(state, mapTitle, secret, numberOfPlayers, slots);
+
+				updateDiscordStatus = false;
+			}
+			else
+			{
+				if (!skirmishMode)
+					DiscordService.UpdatePlayers(numberOfPlayers, slots);
+				DiscordService.UpdateDetails(mapTitle);
```

Comment:
Is there any reason to now not just call `UpdateStatus`, which includes these three fields?

Reply:
The problem with using `UpdateStatus` would be that this resets the invites, and we would need to change it to handle that logic. I dont think its worth doing that change.

Reply:
Ah, i see. Would merging `UpdatePlayers` into `UpdateDetails` also reset them?

Reply:
What i meant above and IMO not worth it.